### PR TITLE
Presenters: init

### PR DESCRIPTION
**Pull request by frizinak**
See [readme](https://github.com/wieni/wmcontroller/tree/feature/presenters) for usage

~~Almost no testing has been done, so don't merge yet.~~

oh... any thoughts?

### CODE REVIEW

**Comment by frizinak** [comment]
@RobinHoutevelts irl: 
>> what's up with the hard to maintain AbstractPresenter (implementation of the huge EntityInterface)

Perhaps instead of piggy backing on EntityPresentedEvent we also dispatch a `SomethingPresentedEvent` with a `setPresenter` method whos argument only needs to implement `setEntity` and `getEntity`, AbstractPresenter then only provides the protected `$entity` property.

```
class SomethingPresentedEvent
{
    public function __construct (HasPresenter $something);

    public function getSomething();

    // Where PresenterInterface no longer extends EntityInterface
    public function setPresenter(PresenterInterface $presenter);
}
```

Pros:
  - arbitrary classes can have arbitrary presenters

Cons:
  - toUrl, id, ... can no longer be assumed

But let's be real, is that con valid? If we're implementing custom classes
with presenters, why would we assume id() / toUrl() in twig?

Interfaces have no meaning in twig anyways.


So yeah, might aswel change this unless someone comes up with a real 'con' ;)

Ideas to rename 'Something' to well... something more meaningful?

File: src/Entity/AbstractPresenter.php [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+{
+    /** @var EntityInterface */
+    protected $entity;
+
+    protected function methodNames($methodName)
+    {
+        return [
+            $methodName,
+            'is' . ucfirst($methodName),
+            'get' . ucfirst($methodName),
```
Comment (frizinak) [inline src/Entity/AbstractPresenter.php]: added, thx

**Comment by spoit** [comment]
OMG I love presenters

**Comment by frizinak** [comment]
removed EntityInterface contract

seems backward compatible for what it's worth

**Comment by RobinHoutevelts** [comment]
> Ideas to rename 'Something' to well... something more meaningful?

Entity isn't that bad a name. I just checked at [Laracasts/Presenter Presenter.php file](https://github.com/laracasts/Presenter/blob/master/src/Laracasts/Presenter/Presenter.php) and even there `entity` is being used. I'd expected `model` :)

Glad the dependency on `EntityInterface` is removed for presenters. 👍

**Comment by frizinak** [comment]
well I went with 'PresentedEvent' and the internal name is `$item`

I'd say merge if we have no issues until tuesday eve, close to impossible to break non-presenter sites anyways.

**Pull request merged**